flexget/components/sites/sites/piratebay.py

```diff
@@ -130,76 +137,80 @@ def search(self, task, entry, config=None):
+            url = '%s/q.php?q=%s%s' % (self.url, quote(query.encode('utf-8')), filter_url)
             logger.debug('Using {} as piratebay search url', url)
             page = task.requests.get(url).content
-            soup = get_soup(page)
-            for link in soup.find_all('a', attrs={'class': 'detLink'}):
-                entry = Entry()
-                entry['title'] = self.extract_title(link)
-                if not entry['title']:
-                    logger.error('Malformed search result. No title or url found. Skipping.')
-                    continue
-                href = link.get('href')
-                if href.startswith('/'):  # relative link?
-                    href = self.url + href
-                entry['url'] = href
-                row = link.parent.parent.parent
-                description = row.find_all('a', attrs={'class': 'detDesc'})
-                if description and description[0].contents[0] == "piratebay ":
-                    logger.debug('Advertisement entry. Skipping.')
-                    continue
-                tds = row.find_all('td')
-                entry['torrent_seeds'] = int(tds[-2].contents[0])
-                entry['torrent_leeches'] = int(tds[-1].contents[0])
-                entry['torrent_availability'] = torrent_availability(
-                    entry['torrent_seeds'], entry['torrent_leeches']
-                )
-                # Parse content_size
-                size_text = link.find_next(attrs={'class': 'detDesc'}).get_text()
-                if size_text:
-                    size = re.search(r'Size (\d+(\.\d+)?\xa0(?:[PTGMK])?i?B)', size_text)
-                    if size:
-                        entry['content_size'] = parse_filesize(size.group(1))
-                    else:
-                        logger.error(
-                            'Malformed search result? Title: "{}", No size? {}',
-                            entry['title'],
-                            size_text,
-                        )
+            json_results = json.loads(page)
+            if len(json_results) == 0:
+                logger.error("Error while searching piratebay for %s." % search_string)
+            for result in json_results:
+                if result['id'] == '0':
+                    # JSON when no results found:
+                    # [{
+                    #    "id":"0",
+                    #    "name":"No results returned",
+                    #    "info_hash":"0000000000000000000000000000000000000000",
+                    #    "leechers":"0",
+                    #    "seeders":"0",
+                    #    "num_files":"0",
+                    #    "size":"0",
+                    #    "username":"",
+                    #    "added":"0",
+                    #    "status":"member",
+                    #    "category":"0",
+                    #    "imdb":"",
+                    #    "total_found":"1"
+                    # }]
+                    break
```

Comment:
Something about this conditional feels like a code smell to me.  Are you trying to avoid indenting the `try` block below?

Reply:
What would you suggest?

Reply:
why `break` and not `continue`? can there be only 1 result in the list?

Reply:
Yes, result['id'] == '0' when no torrent is found or torrent requested does not exist.

Reply:
How about:
```
if result['id'] > 0:
    entries.add(self.json_to_entry(result))
else:
    break
```

Reply:
@ianstalk why add additional nesting and indentation? I much prefer the current way

Reply:
It's definitely a matter of preference, which is why I tried to present it as a suggestion rather than something that needs to change.